### ISSUE TITLE
Add Filter / Location to URL

### ISSUE THREAD

**Issue by mguentner** [MateMonkey/frontend#5]
This could be solved similar to Google Maps, something like

`http://matemonkey.com/map/@23.42,13.37,15/filter=hash`

**Comment by trockenasche** [comment]
I thought more like http://matemonkey.com/map/?München

**Comment by mguentner** [comment]
I suggest using `http://matemonkey.com/map/search/München` for that. However the search will immediately resolve to the latitude/longitude. Currently I don't see an advantage of putting a search term into the URL. Latitude,Longitude and the zoom level are perfect for sharing.

Or do you want to write a Twitter Bot that replies to all "Where can I buy Mate in XY" with `http://matemonkey.com/map/?XY`? :smile:

**Comment by zealot128** [comment]
another thought: having the search pages with city name - or pseudo search pages - is good for Google, too

**Comment by mguentner** [comment]
True, maybe we can implement a hybrid solution. SEO results for Google etc. need to go into a sitemap.xml or similar (like `http://matemonkey.com/map/search/München`).
The real problem is how to change/map everything into one URL.

Example:
- The user visits `http://matemonkey.com/`
- The app uses their geo location and changes the url to `http://matemonkey.com/map/@lat,lon
- The user then selects a dealer (what happens now) `http://matemonkey.com/map/dealer/xy` or `http://matemonkey.com/map/dealer/xy@lat,lon` or `http://matemonkey.com/map/@lat,lon/dealer/xy` 
- The user searches for `Munich`
- What happens now (the dealer might still be selected)  `http://matemonkey.com/search/Munich/@lat,lon/dealer/xy` seems stupid

I think each `http://matemonkey.com/search/` URL should simply resolve to a `http://matemonkey.com/map/@lat,lon URL invalidating the filters and dealer selection.
